src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -185,7 +185,14 @@ winrt::Microsoft::Terminal::Settings::Model::CascadiaSettings CascadiaSettings::
             // We should re-parse, but not re-layer
             resultPtr->_ParseJsonString(resultPtr->_userSettingsString, false);
 
-            _WriteSettings(resultPtr->_userSettingsString);
+            try
+            {
+                _WriteSettings(resultPtr->_userSettingsString);
+            }
+            catch (...)
+            {
+                throw SettingsTypedDeserializationException{ winrt::to_string((RS_(L"WriteSettingsFailed"))) };

```

Comment:
I'm not in love with this one -- SettingsTypedDeserializationException taking a string is because of laziness, not to signal that it is the right way to emit settings error text. I'd rather surface it through the existing mechanisms for settings _warnings_ (which we do have!), if at all possible. Warnings are allocated resource strings in TerminalApp instead of TerminalSettingsModel, admittedly, but  there's already code that handles displaying a dialog with a number of warnings in it :smile:

Reply:
I know this conflicts with what @miniksa asked for, but we've got a long tradition of swallowing exceptions and presenting them to the user as "well something weird happened" here on Terminal. At least sending it up as a warning code (not an error!) DOES NOT discard the entire settings load operation!

Reply:
I agree with the above. Failing to write the file doesn't seem like the kind of thing that makes the entire settings unusable - we've got perfectly valid settings in fact, we just can't update the dynamic profiles. That's not world-ending, we can still run with that.

Reply:
Changed it to a warning now!